src/js/integrations/devicecontext.ts

```diff
@@ -0,0 +1,38 @@
+import { addGlobalEventProcessor, getCurrentHub } from "@sentry/core";
+import { Event, Integration } from "@sentry/types";
+import { NativeModules } from "react-native";
+
+const { RNSentry } = NativeModules;
+
+/** Load device context from native. */
+export class DeviceContext implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = DeviceContext.id;
+  /**
+   * @inheritDoc
+   */
+  public static id: string = "DeviceContext";
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(): void {
+    addGlobalEventProcessor(async (event: Event) => {
+      const self = getCurrentHub().getIntegration(DeviceContext);
+      if (!self) {
+        return event;
+      }
+
+      try {
+        const deviceContexts = await RNSentry.deviceContexts();
+        event.contexts = { ...deviceContexts, ...event.contexts };
```

Comment:
`devideContext` should probably have priority here, so I'd reverse spread operations